test/Dapr.Client.Test/PublishEventApiTest.cs

```diff
@@ -109,11 +111,10 @@ public async Task PublishEventAsync_CanPublishTopicWithNoContent_WithMetadata()
             var task = daprClient.PublishEventAsync(TestPubsubName, "test", metadata);
             httpClient.Requests.TryDequeue(out var entry).Should().BeTrue();
             var request = await GrpcUtils.GetRequestFromRequestMessageAsync<PublishEventRequest>(entry.Request);
-            var jsonFromRequest = request.Data.ToStringUtf8();
 
             request.PubsubName.Should().Be(TestPubsubName);
             request.Topic.Should().Be("test");
-            jsonFromRequest.Should().Be("\"\"");
+            request.Data.Length.Should().Be(0);
```

Comment:
This was a bug identified on inspecting the publish code. Publishing "no content" would result in `""` whereas providing `null` would result in "no content".